spec/unit/resources/hab_build_spec.rb

```diff
@@ -35,16 +22,10 @@
                                     'HAB_NONINTERACTIVE' => 'true')
       )
     end
-
-    it 'uploads the artifact with hab pkg upload' do
-      expect(chef_run).to run_execute('upload-pkg').with(
-        command: '/bin/hab pkg upload  /hab/studios/testproject-teststage-testphase/src/results/purple-frogs-0.7.0-dev-20170403213025-x86_64-linux.hart'
-      )
-    end
   end
 end
 
-describe 'test::upload' do
+describe 'test::publish' do
```

Comment:
I like publish since it mates the habitat semantic of publishing to the depot. I wonder if there isn't a better name for the new action so publish isn't overloaded.